src/GovUk.Education.ExploreEducationStatistics.Admin/Services/ReleaseRepository.cs

```diff
@@ -37,18 +44,133 @@ public ReleaseRepository(ContentDbContext context, IMapper mapper)
+            }
+            else
+            {
+                _statisticsDbContext.Update(existingTheme);
+                _mapper.Map(statsTheme, existingTheme);
+            }
+
+            var statsTopic = new Data.Model.Topic
+            {
+                Id = release.Publication.Topic.Id,
+                Title = release.Publication.Topic.Title,
+                Slug = release.Publication.Topic.Slug,
+                ThemeId = release.Publication.Topic.ThemeId,
+            };
+            var existingTopic = await _statisticsDbContext.Topic.FindAsync(
+                statsTopic.Id);
+            if (existingTopic == null)
+            {
+                await _statisticsDbContext.Topic.AddAsync(statsTopic);
+            }
+            else
+            {
+                _statisticsDbContext.Update(existingTopic);
+                _mapper.Map(statsTopic, existingTopic);
+            }
+
+            var statsPub = new Data.Model.Publication
+            {
+                Id = release.Publication.Id,
+                Title = release.Publication.Title,
+                Slug = release.Publication.Slug,
+                TopicId = release.Publication.TopicId,
+            };
+            var existingPub =
+                await _statisticsDbContext.Publication.FindAsync(
+                    statsPub.Id);
+            if (existingPub == null)
+            {
+                await _statisticsDbContext.Publication.AddAsync(statsPub);
+            }
+            else
+            {
+                _statisticsDbContext.Update(existingPub);
+                _mapper.Map(statsPub, existingPub);
+            }
+
+            var statsRelease = new Data.Model.Release
+            {
+                Id = release.Id,
+                Slug = release.Slug,
+                PublicationId = release.PublicationId,
+                TimeIdentifier = release.TimePeriodCoverage,
+                Year = release.Year,
+                PreviousVersionId = release.PreviousVersionId
+            };
+            var existingRelease =
+                await _statisticsDbContext.Release.FindAsync(statsRelease.Id);
+            if (existingRelease == null)
+            {
+                await _statisticsDbContext.Release.AddAsync(statsRelease);
+            }
+            else
+            {
+                _statisticsDbContext.Update(existingRelease);
+                _mapper.Map(statsRelease, existingRelease);
+            }
+
+            var subject = (await _statisticsDbContext.Subject.AddAsync(new Subject
+            {
+                Filename = subjectFilename,
+                Name = subjectName
+            })).Entity;
+
+            await _statisticsDbContext.ReleaseSubject.AddAsync(new ReleaseSubject
+            {
+                ReleaseId = statsRelease.Id,
+                SubjectId = subject.Id,
+            });
+
+            await _statisticsDbContext.SaveChangesAsync();
+            return subject.Id;
```

Comment:
I see why, but it seems a bit strange that this method `CreateReleaseAndSubjectHierarchy` returns the Subject Id.
I can only think of either returning the Release or the calling code looking up the new Subject by Release Id and Subject name as alternatives. Not sure if you have any better ideas @markhiveit, @ntsim ?

Reply:
I'm happy to leave it as it is - seems like it'd be an easy thing to change later if necessary.